azure-tests/src/main/java/fixtures/azureparametergrouping/AutoRestParameterGroupingTestServiceBuilder.java

```diff
@@ -241,8 +241,15 @@ public AutoRestParameterGroupingTestService buildClient() {
         return client;
     }
 
+    @Generated
+    private void validateBuilder() {
```

Comment:
How often do we expect that developers will be customizing this method? I just wonder if it should be generated if there won't be any validations in it.

Reply:
From Bill, he mostly wants the validation on `endpoint` (which is usually "required" in ClientBuilder).

The other common case that I can think of, is for the credential. E.g. when SDK accept both KeyCredential and TokenCredential, client likely want to validate that one and only one of them is provided in ClientBuilder.

Reply:
https://github.com/Azure/autorest.java/pull/2785

Reply:
I think generating it empty (if that happens) is fine; I assume the compiler is smart enough to elide empty function calls..